src/main/kotlin/dev/kikugie/elytratrims/common/compat/AllTheTrimsCompat.kt

```diff
@@ -33,11 +41,68 @@ object AllTheTrimsCompat {
         color: ARGB,
         cache: (TrimInfo) -> Sprite
     ) {
+        //? if <1.21 {
         val palette = PaletteHelper.getPalette(trim.material.value().ingredient.value())
         for (i in 0 until 8) {
             val sprite = cache(TrimInfo(trim, i))
             if (sprite.missing && !(entity == null || ETRenderer.renderAlways(entity))) continue
             model.render(sprite, matrices, provider, stack, light, palette[i].rgb.withAlpha(color.alpha))
         }
+        //?}
+    }
+
+    fun renderTrimExtended(
+        sprite: Sprite,
+        model: Model,
+        matrices: MatrixStack,
+        provider: VertexConsumerProvider,
+        entity: LivingEntity?,
+        stack: ItemStack,
+        trim: ArmorTrim,
+        light: Int,
+        colour: Int
+    ) {
+        //? if >=1.21 {
+        /*AllTheTrimsClient.getShaderManger().setContext(entity, stack.item)
+
+        val renderer = AllTheTrimsClient.getTrimRenderer()
+        val modelId = if (AllTheTrimsClient.getConfig().overrideExisting) {
+            Identifier.ofVanilla(
+                "trims/models/elytra/%s_%s".format(
+                    trim.pattern.value().assetId().path,
+                    AllTheTrims.DYNAMIC
+                )
+            )
+        } else {
+            renderer.getModelId(sprite)
+        }
+        val renderLayer = if (renderer.useLegacyRenderer(sprite)) {
+            Compat.getElytraTrimsCompat().orElseThrow().elytraTrimRenderLayer
```

Comment:
his reference doesn't exist in 4.0.1 anymore. What should the replacement be?

Reply:
That's just `ElytraTrimsAPI#getElytraLayer`, I'll push